src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -238,6 +238,11 @@ private Task ProcessDataFrameAsync()
                 throw new Http2ConnectionErrorException(Http2ErrorCode.PROTOCOL_ERROR);
             }
 
+            if (_incomingFrame.StreamId > _highestOpenedStreamId)
```

Comment:
This one may need some explaining.

Reply:
Added comments and moved the check to a separate method, since it applies to three different frame types.